sklearn/utils/__init__.py

```diff
@@ -217,6 +217,8 @@ def safe_indexing(X, indices):
                                    indices.dtype.kind == 'i'):
             # This is often substantially faster than X[indices]
             return X.take(indices, axis=0)
+        elif getattr(X, 'format', None) in {'dia', 'bsr', 'coo'}:
```

Comment:
This change was required to get `check_estimator(ResampledTrainer(HalfSampler(), SVC()))` working, but should be tested in `test_utils`.